pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/ForwardIndexConfig.java

```diff
@@ -65,25 +70,32 @@ public int getRawIndexWriterVersion() {
     return _rawIndexWriterVersion;
   }
 
+  @Nullable
+  public DictCompressionType getDictCompressionType() {
```

Comment:
Not sure if this is the right thing, feeling we shouldn't leak abstraction of dictionary to forward index?

Reply:
Dictionary id has some special properties that we can leverage. Changed it to `DictIdCompressionType` to be more clear